### PR TITLE
ST-2: Adding Database stuctures, crud, sessions and connectors

### PR DESCRIPTION
**Pull request by LightningOrionis**
## Changes that have been made:
* Added models
*

### CODE REVIEW

File: src/app/alembic.ini [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+# version_path_separator = :
+# version_path_separator = ;
+# version_path_separator = space
+version_path_separator = os  # default: use os.pathsep
+
+# the output encoding used when revision files
+# are written from script.py.mako
+# output_encoding = utf-8
+
+sqlalchemy.url = driver://user:pass@localhost/dbname
```
Comment (siarheinaseka) [inline src/app/alembic.ini:53]: .env

File: src/app/app/database/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import os
+
+# DATABASE_URL = os.environ.get("DATABASE_URL")
+DATABASE_URL = "postgresql://igor:1234@localhost:5432/spotify_tracker_db"
```
Comment (siarheinaseka) [inline src/app/app/database/__init__.py:4]: .env

File: src/app/app/database/base_class.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from typing import Any
+
+from sqlalchemy.ext.declarative import as_declarative, declared_attr
+
+
+@as_declarative()
+class Base:
+    id: Any
```
Comment (siarheinaseka) [inline src/app/app/database/base_class.py:8]: можно еще добавить created_at, updated_at, is_deleted

File: src/app/app/models/charted_song.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+from sqlalchemy.orm import relationship
+
+from app.database.base import Base
+
+
+class ChartedSong(Base):
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    place = Column(Integer, nullable=False)
+
+    song_id = Column(Integer, ForeignKey("song.id"), nullable=False)
```
Comment (siarheinaseka) [inline src/app/app/models/charted_song.py:11]: nullable=False по деволту же идет, разве нет?

**Review by siarheinaseka — CHANGES_REQUESTED** [review]

File: src/app/app/models/charted_song.py (hunk 4 above)
Comment (LightningOrionis) [reply src/app/app/models/charted_song.py:11]: Если nullable не передана пользователем, nullable = not primary_key

File: src/app/alembic.ini (hunk 1 above)
Comment (LightningOrionis) [reply src/app/alembic.ini:53]: Will be added in ST-3

File: src/app/app/database/__init__.py (hunk 2 above)
Comment (LightningOrionis) [reply src/app/app/database/__init__.py:4]: Will be added in ST-3

File: src/app/app/database/base_class.py (hunk 3 above)
Comment (LightningOrionis) [reply src/app/app/database/base_class.py:8]: Will not be added as there is no need in it

**Pull request merged**